### PR TITLE
fix(ri): Fix RI and modify Massachusetts

### PR DESCRIPTION
**Pull request by flooie**
Merge ri_u and ri_p and adapt it for the new
court website and api.
Update tests

### CODE REVIEW

File: juriscraper/opinions/united_states/state/ri.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        self.request["headers"]["X-RequestDigest"] = request_digest
+        self.request["headers"]["X-Requested-With"] = "XMLHttpRequest"
+        self._request_url_post(self.json_url)
+
+    def _process_html(self) -> None:
+        """Extract content from JSON response
+
+        :return: None
+        """
+        self.fetch_json()
```
Comment (grossir) [inline juriscraper/opinions/united_states/state/ri.py:72]: Getting an error when actually running the script

```
Traceback (most recent call last):
  File "/home/grossir/venvs/juriscraper38/lib/python3.8/site-packages/requests/models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.8/json/__init__.py", line 337, in loads
    raise JSONDecodeError("Unexpected UTF-8 BOM (decode using utf-8-sig)",
json.decoder.JSONDecodeError: Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "sample_caller.py", line 263, in main
    site.parse()
  File "/home/grossir/Desktop/Proyectos/freelaw_repos/juriscraper/juriscraper/AbstractSite.py", line 140, in parse
    self._process_html()
  File "/home/grossir/Desktop/Proyectos/freelaw_repos/juriscraper/juriscraper/opinions/united_states/state/ri.py", line 73, in _process_html
    rows = self.request["response"].json()[-3][self.search_key][
  File "/home/grossir/venvs/juriscraper38/lib/python3.8/site-packages/requests/models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)
```

The content of the json response
```
'\ufeff[\r{\r"SchemaVersion":"15.0.0.0","LibraryVersion":"16.0.10388.20004","ErrorInfo":{\r"ErrorMessage":"Unexpected end of file while parsing Name has occurred. Line 1, position 9896.","ErrorValue":null,"TraceCorrelationId":"bb2638a1-1ad6-a0d3-9c5d-3866ba4ed61b","ErrorStackTrace":"   at System.Xml.XmlTextReaderImpl.Throw(Exception e)\\r\\n   at System.Xml.XmlTextReaderImpl.ParseQName(Boolean isQName, Int32 startOffset, Int32& colonPos)\\r\\n   at System.Xml.XmlTextReaderImpl.ThrowTagMismatch(NodeData startTag)\\r\\n   at System.Xml.XmlTextReaderImpl.ParseEndElement()\\r\\n   at System.Xml.XmlTextReaderImpl.ParseElementContent()\\r\\n   at System.Xml.XsdValidatingReader.Read()\\r\\n   at Microsoft.SharePoint.Client.ClientXmlReader.Read()\\r\\n   at System.Xml.XmlLoader.LoadNode(Boolean skipOverWhitespace)\\r\\n   at System.Xml.XmlLoader.LoadDocSequence(XmlDocument parentDoc)\\r\\n   at System.Xml.XmlDocument.Load(XmlReader reader)\\r\\n   at Microsoft.SharePoint.Client.ClientMethodsProcessor.LoadRequestXml(Stream batchRequest, ClientServiceHost host)\\r\\n   at Microsoft.SharePoint.Client.ClientMethodsProcessor..ctor(Stream batchRequest, ClientServiceHost host, String boundary, IList`1 pendingDisposableQueue)\\r\\n   at Microsoft.SharePoint.Client.ClientRequestServiceImpl.ProcessQuery(Stream inputStream, IList`1 pendingDisposableContainer)","ErrorCode":-1,"ErrorTypeName":"System.Xml.XmlException"\r},"TraceCorrelationId":"bb2638a1-1ad6-a0d3-9c5d-3866ba4ed61b"\r}\r]'
```

File: juriscraper/opinions/united_states/state/mass.py [hunk 2]
```diff
@@ -25,9 +25,19 @@
...
+        self.headers = {
+            "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+            "Sec-Fetch-Site": "none",
+            "Sec-Fetch-Mode": "navigate",
+            "Host": "www.mass.gov",
+            "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.3 Safari/605.1.15",
+            "Accept-Language": "en-US,en;q=0.9",
+            "Sec-Fetch-Dest": "document",
+            "Connection": "keep-alive",
+        }
```
Comment (grossir) [inline juriscraper/opinions/united_states/state/mass.py:40]: I think this won't be used by AbstractSite. You would have to modify `self.request["headers"]`
When changing the headers, I no longer get a 403, but the scraper is not collecting any cases. Maybe the HTML changed?

**Review by grossir — CHANGES_REQUESTED** [review]
I find neither `ri` nor `mass` are working when testing against the live sites

File: juriscraper/opinions/united_states/state/mass.py (hunk 2 above)
Comment (flooie) [reply juriscraper/opinions/united_states/state/mass.py:40]: this is meant to be used with CL 

similar to the Minnesota scraper - it stores it in the site.headers to be used

Comment (flooie) [reply juriscraper/opinions/united_states/state/mass.py:40]: I've updated the changes you suggested.

File: juriscraper/opinions/united_states/state/ri.py (hunk 1 above)
Comment (grossir) [reply juriscraper/opinions/united_states/state/ri.py:72]: I am still getting this error for `ri`. I have tried with different IPs. Not sure what the problem may be. The request goes through when using my browser, but the XML is slightly different

File: juriscraper/opinions/united_states/state/mass.py (hunk 2 above)
Comment (grossir) [reply juriscraper/opinions/united_states/state/mass.py:40]: `mass` is working now

Comment (flooie) [reply juriscraper/opinions/united_states/state/mass.py:40]: thanks @grossir

File: juriscraper/opinions/united_states/state/ri.py (hunk 1 above)
Comment (flooie) [reply juriscraper/opinions/united_states/state/ri.py:73]: perhaps you can slack me the xml you have and I can see if the reverse is true and can be tested.

**Comment by flooie** [comment]
@grossir good to go?

**Review by grossir — APPROVED** [review]

**Pull request merged**